app/scripts/modules/core/src/insight/insightLayout.component.ts

```diff
@@ -1,8 +1,15 @@
 import {InsightFilterStateModel} from './insightFilterState.model';
 import {INSIGHT_NGMODULE} from './insight.module';
+import { Application } from 'core/application';
 
 class InsightLayoutCtrl {
+  public app: Application;
+  public ready = false;
   constructor(public InsightFilterStateModel: InsightFilterStateModel) { 'ngInject'; }
+
+  public $onInit() {
+    this.app.ready().then(() => this.ready = true);
```

Comment:
a lot of this state management stuff could be moved into the router/state machine